include/openssl/core_names.h

```diff
@@ -275,6 +275,7 @@ extern "C" {
 #define OSSL_PKEY_PARAM_PROPERTIES          OSSL_ALG_PARAM_PROPERTIES
 #define OSSL_PKEY_PARAM_DEFAULT_DIGEST      "default-digest" /* utf8 string */
 #define OSSL_PKEY_PARAM_MANDATORY_DIGEST    "mandatory-digest" /* utf8 string */
+#define OSSL_PKEY_PARAM_DIGEST_SUPPORTED    "digest-supported" /* utf8 string */
```

Comment:
You've added a new parameter but it isn't used by anything on the provider side.  I.e. this does exactly nothing.

Reply:
It is supposed to be used by external providers (only). It's the same as `ASN1_PKEY_CTRL_SUPPORTS_MD_NID`. That parameter is also not used by any of the internal openssl providers. It's there for the external providers who impose additional restrictions on digests that could be used with their keys, such as the TPM2.

Reply:
If an external provider supports it, our internal ones must do also.